PXC-4225: Incorrect COMMAND value in I_S.PROCESSLIST for wsrep applie…

kamil-holubicki opened the pull request:
…r threads

https://jira.percona.com/browse/PXC-4225

Problem:
After DML executon or DDL execution after previouw DML, the COMMAND column in I_S.PROCESSLIST shows incorrect value
(Query, instead of Sleep)

Cause:
In normal flow thread's command state is maintained in sql_parse.cc::dispatch_command() functions. PXC flow for replicated events do not use this function, however command state is set to Query in Rows_log_event::do_apply_event() and it remains so.

Solution:
Set proper state of thread's command in Wsrep_applier_service methods called by Galera provider when DDL and DML events are applied.

kamil-holubicki committed:
PXC-4225: Incorrect COMMAND value in I_S.PROCESSLIST for wsrep applier threads

https://jira.percona.com/browse/PXC-4225

Problem:
After DML executon or DDL execution after previouw DML, the COMMAND
column in I_S.PROCESSLIST shows incorrect value
(Query, instead of Sleep)

Cause:
In normal flow thread's command state is maintained in
sql_parse.cc::dispatch_command() functions. PXC flow for replicated
events do not use this function, however command state is set to Query
in Rows_log_event::do_apply_event() and it remains so.

Solution:
Set proper state of thread's command in Wsrep_applier_service methods
called by Galera provider when DDL and DML events are applied.

kamil-holubicki commented:
https://pxc.cd.percona.com/job/pxc-8.0-pipeline-parallel-mtr/1677/

venkatesh-prasad-v commented on sql/wsrep_high_priority_service.cc:
```diff
@@ -483,6 +484,7 @@ int Wsrep_high_priority_service::apply_toi(const wsrep::ws_meta &ws_meta,
            (long long)wsrep_thd_trx_seqno(m_thd));
   WSREP_DEBUG("%s", m_thd->wsrep_info);
   thd_proc_info(thd, m_thd->wsrep_info);
+  thd->set_command(COM_SLEEP);
```
Can you please check if the same fix is required for NBO as well?

kamil-holubicki replied:
NBO was not affected in the way reported in the PXC-4225, but I improved thread state reporting for it.

kamil-holubicki committed:
PXC-4225: Incorrect COMMAND value in I_S.PROCESSLIST for wsrep applier threads

https://jira.percona.com/browse/PXC-4225

Improved thread status reporting during NBO.

venkatesh-prasad-v commented on sql/wsrep_high_priority_service.cc:
```diff
@@ -860,6 +890,28 @@ int Wsrep_applier_service::apply_nbo_begin(const wsrep::ws_meta &ws_meta,
+  };);
+
+  /* WSREP applier thread created NBO background worker
+     and is read for processing of next writeset. */
```
```suggestion
     and is ready for processing next writesets. */
```

venkatesh-prasad-v commented on sql/wsrep_high_priority_service.cc:
```diff
@@ -810,13 +830,23 @@ int Wsrep_applier_service::apply_nbo_begin(const wsrep::ws_meta &ws_meta,
 
     ret = trans_commit(thd);
 
+    /* This tread is going to be removed soon and will be not visible in
```
```suggestion
    /* This thread is going to be removed soon and will be not visible in
```

kamil-holubicki replied:
fixed
